packages/template-retail-react-app/app/hooks/use-wish-list.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useCustomerProductLists, useShopperCustomersMutation} from 'commerce-sdk-react-preview'
+import {useCurrentCustomer} from './use-current-customer'
+
+export const useWishList = ({listId = ''} = {}) => {
+    const {data: customer} = useCurrentCustomer()
+    const {isRegistered, customerId} = customer
+    const createCustomerProductList = useShopperCustomersMutation('createCustomerProductList')
+    const {
+        data: productLists,
+        isLoading: isProductListsLoading,
+        error: productListsError
+    } = useCustomerProductLists(
+        {
+            parameters: {customerId}
+        },
+        {
+            onError: (e) => {
+                console.error('e', e)
+            },
+            onSuccess: (data) => {
+                if (!data.total) {
+                    createCustomerProductList.mutate({
+                        parameters: {customerId},
+                        // we only use one type of product list for now
+                        body: {type: 'wish_list'}
+                    })
+                }
+            },
+            // only registered user can have product lists
+            enabled: isRegistered
+        }
+    )
+    const wishLists = productLists?.data?.filter((list) => list.type === 'wish_list') || []
+    const currentWishlist = wishLists?.data?.find((list) => list.id === listId) || wishLists[0]
+    return {
+        error: productListsError,
+        isLoading: isProductListsLoading,
+        wishlist: currentWishlist
+    }
```

Comment:
I'd love to keep the contract the same as react-query. This applies to all data fetching related custom hooks. 
```js
const query = useCustomerProductLists(...)

const wishlist = query.data?.filter(...)

// important to keep the pattern for DX reasons
// the developers will find this to be easier to use
return {
   ...query,
   data: wishlist
}
```